lib/common.js

```diff
@@ -459,20 +459,22 @@ function formatQueryValue(key, value, stringFormattingFn) {
     case value instanceof RegExp:
       break
     case _.isArray(value): {
-      const tmpArray = new Array(value.length)
-      for (let i = 0; i < value.length; ++i) {
-        tmpArray[i] = formatQueryValue(i, value[i], stringFormattingFn)[1]
-      }
-      value = tmpArray
+      value = value.map(function(val, idx) {
+        return formatQueryValue(idx, val, stringFormattingFn)[1]
+      })
       break
     }
     case _.isObject(value): {
-      const tmpObj = {}
-      _.forOwn(value, function(subVal, subKey) {
-        const subPair = formatQueryValue(subKey, subVal, stringFormattingFn)
-        tmpObj[subPair[0]] = subPair[1]
-      })
-      value = tmpObj
+      value = Object.keys(value).reduce(function(acc, subKey) {
```

Comment:
`Object.entries` is probably better suited here.

```js
Object.entries(value).reduce(function(acc, [subVal, subKey]) {
  const subPair = formatQueryValue(subKey, subVal, stringFormattingFn)
  acc[subPair[0]] = subPair[1]

  return acc
}, {})
```

Reply:
Fixed with https://github.com/nock/nock/pull/1598/commits/6a785a816dc1382572e6c7a67ac3a4e37378d880.